enarx-keep-sev/src/vm.rs

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::x86_64::*;
+
+use std::error::Error;
+
+use kvm_bindings::kvm_userspace_memory_region as MemoryRegion;
+use kvm_ioctls::{Kvm, VmFd};
+use x86_64::structures::paging::page_table::PageTableFlags;
+use x86_64::PhysAddr;
+
+const DEFAULT_MEM_SIZE: usize = units::bytes!(1; GiB);
+
+pub struct VirtualMachine {
+    kvm: Kvm,
+    fd: VmFd,
+    address_space: Vec<MemoryRegion>,
+    mmap_handles: Vec<mmap::Unmap>,
+}
+
+impl VirtualMachine {
+    pub fn new() -> Result<Self, Box<dyn Error>> {
```

Comment:
Aren't all the errors `std::io::Error`?

Reply:
Many of them are. The KVM stuff in particular returns `kvm_ioctls::Error`

Reply:
Ugh... That's why I used `std::io::Error` in `ketuvim`. Everything that comes from an `errno` should be `std::io::Error`, IMHO.

Please file a debt issue for this and I'll approve.

Reply:
Filed under https://github.com/enarx/enarx/issues/556